x/lockup/keeper/lock_test.go

```diff
@@ -45,6 +45,67 @@ func (suite *KeeperTestSuite) TestBeginUnlocking() { // test for all unlockable
 	suite.Require().NotEqual(locks[0].IsUnlocking(), false)
 }
 
+func (suite *KeeperTestSuite) TestBeginForceUnlock() {
+	// coins to lock
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+
+	testCases := []struct {
+		name             string
+		coins            sdk.Coins
+		unlockCoins      sdk.Coins
+		expectSameLockID bool
+	}{
+		{
+			name:             "new lock id is returned if the lock was split",
+			coins:            coins,
+			unlockCoins:      sdk.Coins{sdk.NewInt64Coin("stake", 1)},
+			expectSameLockID: false,
+		},
+		{
+			name:             "same lock id is returned if the lock was not split",
+			coins:            coins,
+			unlockCoins:      sdk.Coins{},
+			expectSameLockID: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.SetupTest()
```

Comment:
For each test case, let's create a separate environment for it with suite.`Run(...). Similar to:
https://github.com/osmosis-labs/osmosis/blob/62869348e43946f49c3886f5b1aa44d67ebe721a/x/gamm/keeper/pool_service_test.go#L302-L303

I know this is not done in most of the test cases in the file. However, this is the correct way that enables a separate test environment for each table-driven test case.